app/src/main/java/net/gsantner/opoc/util/FileUtils.java

```diff
@@ -177,8 +209,13 @@ public static byte[] readCloseBinaryStream(final InputStream stream) {
         return baos.toByteArray();
     }
 
-    public static boolean writeFile(final File file, final byte[] data) {
+    public static boolean writeFile(final File file, final byte[] data, boolean withBOM) {
```

Comment:
when you agree with the metadata comment, you could pass it in here again and so withBom information gets auto extracted from that `Map`.